### PR TITLE
refactors and big improvements

### PR DESCRIPTION
**Pull request by Mooy1**
## Changes
Moved some tickers and listeners to AlienManger and WorldManager. cleaned up some stuff.

## Related Issues
x

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break world generation, mob spawning, and the like.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

**Review by Seggan — APPROVED** [review]
just a few things

**Review by Seggan — APPROVED** [review]
just a few things

File: src/main/java/io/github/addoncommunity/galactifun/api/aliens/Alien.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+    protected void onDamage(EntityDamageEvent e) { }
+
+    /**
+     * Gets the AI of the Alien. The returned map is a map of a mob goal and its priority
+     *
+     * @return a map of the Alien's goals, or an empty map for default AI
+     */
+    protected Map<Goal<T>, Integer> getGoals() {
+        return Map.of();
```
[the remark this answers is not in the document]
Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/api/aliens/Alien.java:134]: Just did it so it wouldn't initialize a new hashmap every time something spawns

File: src/main/java/io/github/addoncommunity/galactifun/Galactifun.java [hunk 2]
```diff
@@ -50,6 +63,18 @@ public void onEnable() {
         ));
     }
 
+    @Override
+    protected void disable() {
+        // todo make better
```
[the remark this answers is not in the document]
Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/Galactifun.java:68]: The removing bossbars thing, theres gotta be a cleaner way to do it i think.

**Pull request merged**